core/domain/suggestion_services_test.py

```diff
@@ -2084,6 +2094,229 @@ def test_create_translation_contribution_stats_from_model(self) -> None:
+        topic_id = topic_fetchers.get_new_topic_id()
+        self.save_new_topic(
+            topic_id, 'topic_admin', name='Topic1',
+            abbreviated_name='topic-three', url_fragment='topic-three',
+            description='Description',
+            canonical_story_ids=[],
+            additional_story_ids=[],
+            uncategorized_skill_ids=[skill_id],
+            subtopics=[], next_subtopic_id=1)
+        suggestion_change = {
+            'cmd': (
+                question_domain
+                .CMD_CREATE_NEW_FULLY_SPECIFIED_QUESTION),
+            'question_dict': {
+                'question_state_data': self._create_valid_question_data(
+                    'default_state').to_dict(),
+                'language_code': 'en',
+                'question_state_data_schema_version': (
+                    feconf.CURRENT_STATE_SCHEMA_VERSION),
+                'linked_skill_ids': ['skill_1'],
+                'inapplicable_skill_misconception_ids': ['skillid12345-1']
+            },
+            'skill_id': skill_id,
+            'skill_difficulty': 0.3
+        }
+        suggestion = suggestion_services.create_suggestion(
+            feconf.SUGGESTION_TYPE_ADD_QUESTION,
+            feconf.ENTITY_TYPE_SKILL, skill_id, 1,
+            self.author_id, suggestion_change, 'test description')
+        suggestion_services.update_question_contribution_stats(
+            suggestion, self.author_id, skill_id, True
+        )
+
+        question_contribution_stats_model = (
+            suggestion_models.QuestionContributionStatsModel.get(
+                self.author_id, topic_id
+            )
+        )
+        self.assertEqual(
+            question_contribution_stats_model.submitted_questions_count,
+            1
+        )
+        self.assertEqual(
+            question_contribution_stats_model.accepted_questions_count,
+            0
+        )
+
+        suggestion.status = suggestion_models.STATUS_ACCEPTED
+        suggestion_services.update_question_review_stats(
+            suggestion, self.reviewer_id, skill_id, True
+        )
+
+        question_review_stats_model = (
+            suggestion_models.QuestionReviewStatsModel.get(
+                self.reviewer_id, topic_id
+            )
+        )
+        question_contribution_stats_model = (
+            suggestion_models.QuestionContributionStatsModel.get(
+                self.author_id, topic_id
+            )
+        )
+
+        self.assertEqual(
+            question_contribution_stats_model.submitted_questions_count,
+            1
+        )
+        self.assertEqual(
+            question_contribution_stats_model.accepted_questions_count,
+            1
+        )
+        self.assertEqual(
+            question_review_stats_model.accepted_questions_count,
+            1
+        )
+        self.assertEqual(
+            (
+                question_review_stats_model
+                .reviewed_questions_count
+            ),
+            1
+        )
+
```

Comment:
In general, these tests are pretty minimal. Try to cover different-length submissions, rejected questions, and other cases.

Reply:
I'm having trouble seeing those other cases (e.g. I can't find rejected questions, etc.).  The test names don't help because they just say "test this function". 

Can you either clarify what's being tested in the test names, or leave comments throughout the test to explain what cases/behaviours specific parts of the test are covering?

Reply:
I added more cases for acceptances, rejects and reviewer edits. Furthermore explained some complex tests cases by commenting.